tags/context.go

```diff
@@ -0,0 +1,49 @@
+package grpc_ctxtags
+
+import "context"
+
+type ctxMarker struct{}
+
+var (
+	// ctxMarkerKey is the Context value marker used by *all* logging middleware.
+	// The logging middleware object must interf
+	ctxMarkerKey = &ctxMarker{}
+)
+
+// Tags is the struct used for storing request tags between Context calls.
+// This object is *not* thread safe, and should be handled only in the context of the request.
+type Tags struct {
+	values map[string]interface{}
+}
+
+// Set sets the given key in the metadata tags.
+func (t *Tags) Set(key string, value interface{}) *Tags {
```

Comment:
Your `Set` method both updates the map, and returns a tags object that you're expecting users to use. Given that updates work, it seems likely that users are going to call `Set` without using the return value (I can't even spot where you specify they shouldn't). That will work until they happen to be in a context where `Extract` returns a completely new tag object, at which point tags will suddenly be silently dropped.